fabric-structures-v0/src/main/java/net/fabricmc/fabric/api/structures/v0/FabricStructures.java

```diff
@@ -0,0 +1,48 @@
+package net.fabricmc.fabric.api.structures.v0;
+
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.gen.GenerationStep;
+import net.minecraft.world.gen.feature.FeatureConfig;
+import net.minecraft.world.gen.feature.StructureFeature;
+
+/**
+ * API that hooks into the internal structure generation step logic
+ */
+public final class FabricStructures {
+	private FabricStructures() { }
+
+	/**
+	 * Adds a structure to the given generation step
+	 *
+	 * @param structure The structure to add to the step map. Must not be null
+	 */
+	public static void setGenerationStep(FabricStructure structure) {
+		// Put if absent was used to avoid vanilla overrides
+		StructureFeature.STRUCTURE_TO_GENERATION_STEP.putIfAbsent(structure, structure.generationStep());
+	}
+
+	/**
+	 * Appends a structure to the internal structure hashmap for serialization
+	 *
+	 * Now if users want to change the name of their structure
+	 *
+	 * @param structure The structure to be appended
+	 */
+	public static void appendStructure(FabricStructure structure) {
```

Comment:
Maybe `register`/`add`? Also is this ever going to be used apart from the actual `register` below

Reply:
Ideally it never would, I just wasn't sure how struct we wanted to be with specific implementation

Reply:
Resolved in a6b4507